frontend/src/pages/chat/components/chat-layout/chat-layout.tsx

```diff
@@ -1,34 +1,62 @@
-import { UserRole } from '#libs/enums/enums.js';
-import { useCallback } from '#libs/hooks/hooks.js';
+import {
+  useAppDispatch,
+  useAppSelector,
+  useCallback,
+  useParams,
+} from '#libs/hooks/hooks.js';
+import { type UserAuthResponseDto } from '#packages/users/users.js';
 import {
   ChatFooter,
   ChatHeader,
   ChatMessage,
 } from '#pages/chat/components/components.js';
-import { MOCK_MESSAGES } from '#pages/chat/libs/constants/constants.js';
+import {
+  EMPTY_ARRAY_LENGTH,
+  MOCK_MESSAGES,
+} from '#pages/chat/libs/constants/constants.js';
+import { type ChatInputValue } from '#pages/chat/libs/types/types.js';
+import { actions as chatActions } from '#slices/chats/chats.js';
 
 import styles from './styles.module.scss';
 
 const ChatLayout: React.FC = () => {
-  const handleSend = useCallback((): void => {
-    // TODO: dispatch redux action to send message
-  }, []);
+  const { id } = useParams<{ id: string }>();
+  const dispatch = useAppDispatch();
+  const { currentChatMessages, authenticatedUser } = useAppSelector(
+    ({ chats, auth }) => {
+      return {
+        currentChatMessages: chats.currentChatMessages,
+        authenticatedUser: auth.authenticatedUser as UserAuthResponseDto,
+      };
+    },
+  );
+
+  const handleSend = useCallback(
+    ({ message }: ChatInputValue): void => {
+      if (!id && currentChatMessages.length === EMPTY_ARRAY_LENGTH) {
+        void dispatch(chatActions.createChat({ message }));
+      }
+      // TODO: dispatch redux action to send message
+    },
+    [dispatch, currentChatMessages.length, id],
+  );
 
   return (
     <>
       <ChatHeader />
       <div className={styles['chat-body']}>
-        {MOCK_MESSAGES.map((item) => {
-          return (
-            <ChatMessage
-              key={item.id}
-              item={item}
-              isSender={item.sender === UserRole.USER}
-            />
-          );
-        })}
+        {Boolean(id) &&
+          MOCK_MESSAGES.map((item) => {
+            return (
+              <ChatMessage
+                key={item.id}
+                message={item.message}
+                isSender={item.senderId === authenticatedUser.id}
+              />
+            );
+          })}
       </div>
-      <ChatFooter onSend={handleSend} />
+      <ChatFooter key={id} onSend={handleSend} />
```

Comment:
```suggestion
      <ChatFooter onSend={handleSend} />
```

Reply:
It was made on purpose for clearing a chat input after changing routes. If I remove this then the chat input will not clear its value. I am updating the chat footer every time we change the route. It is just a simple method of doing so)

Reply:
here is the link to official documentation about this: https://react.dev/reference/react/useState#resetting-state-with-a-key